app/actions/project.js

```diff
@@ -13,9 +13,10 @@ export const ADD_REQUEST = 'ADD_REQUEST';
 export const UPDATE_REQUEST = 'UPDATE_REQUEST';
 export const DELETE_REQUEST = 'DELETE_REQUEST';
 export const EXECUTE_REQUEST = 'EXECUTE_REQUEST';
+export const LOG_REQUEST = 'LOG_REQUEST';
 
 export const Request = Record({
-  id: undefined,
+  id: randomId(),
```

Comment:
The way I see it the intent of the id is to refer to the request regardless of it having been persisted. For example through routing.